lesson-7/todos/src/components/editNotesPage/EditNoteModal.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import Modal from 'react-modal'
+import '../../styles/EditNoteModal.css'
+
+class EditNoteModal extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      modalIsOpen: false,
+      note: {
+        uuid: props.noteUuid,
+        text: props.text,
+        title: props.title,
+      },
+    }
+  }
+
+  toggleModal = () => {
+    this.setState({
+      modalIsOpen: !this.state.modalIsOpen,
+    })
+  }
+
+  handleChange = field => e => {
+    let note = this.state.note
+    note[field] = e.target.value
+    this.setState({ note })
+  }
+
+  handleFormSubmit = e => {
+    e.preventDefault()
+    this.props.editNote(this.state.note)
+    this.toggleModal()
+  }
+
+  render() {
+    const { modalIsOpen } = this.state
+    return (
+      <div className="EditNoteModal">
+        <a onClick={this.toggleModal}>edit</a>
+        <Modal
+          isOpen={modalIsOpen}
+          onRequestClose={this.toggleModal}
+          className="EditNoteModal-modal-window"
+          overlayClassName="EditNoteModal-modal-overlay"
+        >
+          <h2>Edit a note</h2>
+          <form onSubmit={this.handleFormSubmit}>
+            <div>
+              <label htmlFor="title">Title</label>
+              <input
+                type="text"
+                id="title"
+                value={this.state.note.title}
+                onChange={this.handleChange('title')}
```

Comment:
Bad practice; this will always create new instance which makes it fail the instance equality test which forces React to rerender even though the behavior is always the same.

https://medium.freecodecamp.org/why-arrow-functions-and-bind-in-reacts-render-are-problematic-f1c08b060e36